rasa/utils/tensorflow/temp_keras_modules.py

```diff
@@ -0,0 +1,258 @@
+import copy
+from typing import (
+    List,
+    Dict,
+    Union,
+    Optional,
+    Any,
+)
+
+import numpy as np
+import tensorflow as tf
+
+from tensorflow.python.keras.callbacks import Callback
+from tensorflow.python.keras.engine import training
+from tensorflow.python.keras.engine import training_utils
+from tensorflow.python.keras.engine import data_adapter
+from tensorflow.python.keras.utils import version_utils
+from tensorflow.python.eager import context
+from tensorflow.python.keras.engine.data_adapter import DataHandler
+
+
+# noinspection PyMethodOverriding
+class TmpKerasModel(tf.keras.models.Model):
+    """Temporary solution. Keras model that uses a custom data adapter inside fit."""
+
+    # TODO
+    #  we don't need this anymore once
+    #  https://github.com/tensorflow/tensorflow/pull/45338
+    #  is merged and released
```

Comment:
Can we also add an attribution comment something like - `This code is adapted from <link>`